src/libraries/Microsoft.Extensions.DependencyInjection/src/ServiceLookup/CallSiteFactory.cs

```diff
@@ -360,6 +368,32 @@ void AddCallSite(ServiceCallSite callSite, int index)
             {
                 callSiteChain.Remove(serviceIdentifier);
             }
+
+            static bool KeysMatch(object? lookupKey, object? descriptorKey)
```

Comment:
```suggestion
            static bool KeysMatchEnumerable(object? lookupKey, object? descriptorKey)
```

Nit: I suggested this before noticing this is nested, so feel free to ignore.